README.md

```diff
@@ -186,3 +186,161 @@ If none package suits scope of new changes, we need to create new package by cre
 Webhooks are enabled to trigger releases on travis from comment on merged PR. If you are member of group responsible for releases you can add new commnent to merged PR `Release minor`, `Release bugfix` or `Release` in order to trigger new release.
 
 You can also draft a release by adding label `release` or `release minor` and once this PR is merged new release will be triggered.
+
+## Typescript guidelines
+
+Typescript build has been finally introduced to the repository. All modules will be eventually refactored into typescript. When refactoring, or creating new modules, please follow these instructions when creating types:
+
+### React components
+
+1. All component props must be an interface and must be exported. This is required for consuming apps to use features like `omit` or `extends` when developing with typescript. A nice example is generating JSX from array structure, but for some reseason extension/exclustion of the interface is required.
```

Comment:
Do we want to use `interface`s? Consumers can extend types as well (either with [intersection-types](https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types) or since TS 2.2 they can use interface and directly use `extends` the type). I think that using `type` gives us a bit more flexibility (type can't extend interface, but interface can extend type) and looks like many upstream projects prefer types as well (only for some specific usecases they export `interface`).